src/Winton.Extensions.Configuration.Consul/ConsulConfigurationProvider.cs

```diff
@@ -22,7 +22,7 @@ internal sealed class ConsulConfigurationProvider : ConfigurationProvider
         {
             if (source.Parser == null)
             {
-                throw new ArgumentNullException(nameof(source.Parser));
+                throw new ArgumentException($"{nameof(source.Parser)} is null", nameof(source));
```

Comment:
Can you point to any documentation for why this is preferred / why `ArgumentNullException` usage generates a warning? I thought it might be related to nullable reference types in C# 8 but I can't find any references to this online.

Reply:
After much head-scratching, I discovered that the warning I was seeing was a result of having the [SonarLint Plugin](https://www.sonarlint.org/visualstudio/) installed in Visual Studio. The warnings it gave me are:
- [General exceptions should never be thrown](https://rules.sonarsource.com/csharp/RSPEC-112) (which I ignored in the `<NoWarn>` project property)
- [Parameter names used into ArgumentException constructors should match an existing one](https://rules.sonarsource.com/csharp/RSPEC-3928) (which I fixed here)

I'm not happy with the way this plugin works. I installed it as we are rolling out  `SonarQube` at the place I work. But I had assumed that you have to opt-in for having it run on on a per-project or per-solution basis. Unfortunately that is not the case and I will now be uninstalling it and cursing under my breath.